metaworld/envs/mujoco/sawyer_xyz/v2/sawyer_window_close_v2.py

```diff
@@ -88,38 +101,54 @@ def reset_model(self):
         self.sim.model.body_pos[self.model.body_name2id(
             'window'
         )] = self.obj_init_pos
+        self.window_handle_pos_init = (self._get_pos_objects()
+            + np.array([0.2, 0., 0.]))
         self.data.set_joint_qpos('window_slide', 0.2)
 
+
         return self._get_obs()
 
     def _reset_hand(self):
         super()._reset_hand()
-        self.reachCompleted = False
+        self.init_tcp = self.tcp_center
 
     def compute_reward(self, actions, obs):
         del actions
+        obj = self._get_pos_objects()
+        tcp = self.tcp_center
+        target = self._target_pos.copy()
+        
+        target_to_obj = (obj[0] - target[0])
+        target_to_obj = np.linalg.norm(target_to_obj)
+        target_to_obj_init = (self.window_handle_pos_init[0] - target[0])
+        target_to_obj_init = np.linalg.norm(target_to_obj_init)
+
+        in_place = reward_utils.tolerance(
+            target_to_obj,
+            bounds=(0, self.TARGET_RADIUS),
+            margin=abs(target_to_obj_init - self.TARGET_RADIUS),
+            sigmoid='long_tail',
+        )
 
-        objPos = obs[3:6]
-
-        rightFinger, leftFinger = self._get_site_pos('rightEndEffector'), self._get_site_pos('leftEndEffector')
-        fingerCOM  =  (rightFinger + leftFinger)/2
-
-        pullGoal = self._target_pos
-
-        pullDist = np.abs(objPos[0] - pullGoal[0])
-        reachDist = np.linalg.norm(objPos - fingerCOM)
-
-        self.reachCompleted = reachDist < 0.05
-
-        c1 = 1000
-        c2 = 0.01
-        c3 = 0.001
-        reachRew = -reachDist
-
-        if self.reachCompleted:
-            pullRew = 1000*(self.maxPullDist - pullDist) + c1*(np.exp(-(pullDist**2)/c2) + np.exp(-(pullDist**2)/c3))
-        else:
-            pullRew = 0
-        reward = reachRew + pullRew
 
-        return [reward, reachDist, None, pullDist]
+        handle_radius = 0.02
```

Comment:
What does handle radius represent for the window? Is it like the handle width, or the distance between the handle and the glass panes?

Reply:
it represents the distance that the gripper can be away from the handle, and get a reach reward of 1